File: src/main/java/io/quarkiverse/test/junit/mockk/InjectMock.java
package io.quarkiverse.test.junit.mockk;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * When used on a field of a test class, the field becomes a Mockk mock,
 * that is then used to mock the normal scoped bean which the field represents
 */
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
public @interface InjectMock {

    boolean relaxed() default false;

    boolean relaxUnitFun() default false;

}

File: src/main/kotlin/io/quarkiverse/test/junit/mockk/InjectMock.kt
package io.quarkiverse.test.junit.mockk



/**
 * When used on a field of a test class, the field becomes a Mockk mock,
 * that is then used to mock the normal scoped bean which the field represents
 */
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
annotation class InjectMock(val relaxed: Boolean = false, val relaxUnitFun: Boolean = false)
